enterprise_subsidy/apps/api/v2/tests/test_transaction_views.py

```diff
@@ -0,0 +1,560 @@
+        # user read the transactions for subsidy_3
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_3_uuid])
+        response = self.client.get(url)
+
+        assert response.status_code == status.HTTP_200_OK
+        list_response_data = response.json()["results"]
+        response_uuids = [tx["uuid"] for tx in list_response_data]
+        expected_response_uuids = [
+            APITestBase.subsidy_3_transaction_1_uuid,
+            APITestBase.subsidy_3_transaction_2_uuid,
+        ]
+        self._prepend_initial_transaction_uuid(self.subsidy_3_uuid, expected_response_uuids)
+        self.assertEqual(response_uuids, expected_response_uuids)
+
+    def test_learner_cannot_use_admin_list(self):
+        """
+        Tests that an authenticated learner, even one with role-based access
+        to a given customer, gets a 403 when requesting admin list access to transactions therein.
+        """
+        self.set_up_learner()
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_1.uuid])
+
+        response = self.client.get(url)
+
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    @ddt.data('admin', 'operator')
+    def test_list_no_matching_subsidy_uuid_error_response(self, role):
+        """
+        Tests that an authenticated admin or operator receives a 403 response
+        when requesting a subsidy uuid that does not exist.
+        """
+        if role == 'admin':
+            self.set_up_admin()
+        if role == 'operator':
+            self.set_up_operator()
+
+        url = reverse("api:v2:transaction-admin-list-create", args=[uuid.uuid4()])
+
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+@ddt.ddt
+class TransactionAdminCreateViewTests(APITestBase):
+    """
+    Test transaction creation operations via the transaction-admin-list-create view.
+    """
+    @classmethod
+    def setUpClass(cls):
+        """
+        We only need to setup subsidies for the creation tests.
+        """
+        APITestMixin.setUpClass()
+        cls._setup_subsidies()
+
+    def tearDown(self):
+        """
+        Deletes any transactions with a non-null lms_user_id (i.e. everything
+        except the initializing transactions for the subsidies).
+        """
+        Transaction.objects.exclude(lms_user_id__isnull=True).delete()
+
+    @ddt.data('learner', 'admin')
+    def test_learners_and_admins_cannot_create_transactions(self, role):
+        """
+        Neither learner or admin roles should provide the ability to create transactions.
+        """
+        if role == 'admin':
+            self.set_up_admin()
+        if role == 'learner':
+            self.set_up_learner()
+
+        url = reverse("api:v2:transaction-admin-list-create", args=[self.subsidy_1.uuid])
+
+        response = self.client.post(url, {'anything': 'goes'})
```

Comment:
IMO the post parameters for a test like this should be valid to narrow the scope of what's being tested